app/javascript/article-form/components/ImageUploader.jsx

```diff
@@ -35,136 +36,333 @@ const ImageIcon = () => (
     uploadingImage: false,
-  };
-
-  onUploadError = (error) => {
-    this.setState({
-      insertionImageUrls: [],
-      uploadError: true,
-      uploadErrorMessage: error.message,
-      uploadingImage: false,
+  });
+
+  const {
+    uploadingImage,
+    showImageCopiedMessage,
+    uploadErrorMessage,
+    uploadError,
+    insertionImageUrls,
+  } = state;
+
+  let imageMarkdownInput = null;
+
+  function onUploadError(error) {
+    dispatch({
+      type: 'upload_error',
+      payload: { errorMessage: error.message },
     });
-  };
+  }
 
-  copyText = () => {
-    this.imageMarkdownInput = document.getElementById(
+  function copyText() {
+    imageMarkdownInput = document.getElementById(
       'image-markdown-copy-link-input',
     );
 
     if (isNativeAndroid()) {
-      AndroidBridge.copyToClipboard(this.imageMarkdownInput.value);
-      this.setState({ showImageCopiedMessage: true });
+      AndroidBridge.copyToClipboard(imageMarkdownInput.value);
+      dispatch({
+        type: 'show_copied_image_message',
+        // payload: true,
+      });
+      // setShowImageCopiedMessage(true);
     } else if (isClipboardSupported()) {
       navigator.clipboard
-        .writeText(this.imageMarkdownInput.value)
+        .writeText(imageMarkdownInput.value)
         .then(() => {
-          this.setState({ showImageCopiedMessage: true });
+          dispatch({
+            type: 'show_copied_image_message',
+            // payload: true,
+          });
+          // setShowImageCopiedMessage(true);
         })
         .catch((_err) => {
-          this.execCopyText();
+          execCopyText();
         });
     } else {
-      this.execCopyText();
+      execCopyText();
     }
-  };
+  }
 
-  handleInsertionImageUpload = (e) => {
+  function handleInsertionImageUpload(e) {
     const { files } = e.target;
-
+    // debugger;
     if (files.length > 0 && validateFileInputs()) {
       const payload = { image: files };
-
-      this.setState({
-        uploadError: false,
-        uploadErrorMessage: null,
-        uploadingImage: true,
-        insertionImageUrls: [],
-        showImageCopiedMessage: false,
+      // debugger;
+      dispatch({
+        type: 'uploading_image',
+        // payload:
       });
 
-      generateMainImage(
-        payload,
-        this.handleInsertImageUploadSuccess,
-        this.onUploadError,
-      );
+      // setUploadError(false);
+      // setUploadErrorMessage(null);
+      // setUploadingImage(true);
+      // setInsertionImageUrls([]);
+      // setShowImageCopiedMessage(false);
+
+      generateMainImage(payload, handleInsertImageUploadSuccess, onUploadError);
     }
-  };
+  }
 
-  handleInsertImageUploadSuccess = (response) => {
-    this.setState({
-      uploadingImage: false,
-      insertionImageUrls: response.links,
+  function handleInsertImageUploadSuccess(response) {
+    dispatch({
+      type: 'upload_image_success',
+      payload: { responseLink: response.links },
```

Comment:
The test was failing because the insertion image URLS were not being set.

```suggestion
      payload: { insertionImageUrls: response.links },
```